app/models/miq_queue.rb

```diff
@@ -544,6 +598,16 @@ def self.get_worker(task_id)
 
   private
 
+  def self.update_with_targets!(msg, save_options)
+    targets = save_options[:targets]
+
+    sql_target_array = targets.map { |target| "#{connection.quote(target.to_json)}::text" }.join(",")
+    set_clauses      = ["targets = targets || ARRAY[#{sql_target_array}]"]
+    set_clauses << sanitize_sql_for_assignment(save_options.except(:targets))
+
+    where(:id => msg.id).update_all(set_clauses.join(", "))
```

Comment:
@Fryguy seems like this doesn't set the ```lock_version``` column, should I do that manually and check it? I think I should...